### ISSUE TITLE
Codeql to detect CORS misconfiguration in go webapp

### ISSUE THREAD

**Issue by AdmiralGaust** [github/codeql-go#576]
Here is how a sample go code vulnerable to CORS misconfiguration looks like

```
import(
"github.com/go-chi/cors"
)

var corsOpts = cors.Options{
	AllowedOrigins:   []string{"*"},
	AllowCredentials: true
}
```

It does not validate the origin header and allows authenticated cross-origin requests. I was testing codeql on a popular GitHub repo and unfortunately, it was not able to detect it using lgtm.com website. So I thought this query can be added to codeql for detecting this kind of bugs.

**Comment by smowton** [comment]
Did you try using the experimental query https://github.com/github/codeql-go/blob/main/ql/src/experimental/CWE-942/CorsMisconfiguration.ql ?

**Comment by AdmiralGaust** [comment]
Yeah, tried this but it produces zero results.-

**Comment by intrigus-lgtm** [comment]
This is intentional.

The code for the case your interested in has been removed in https://github.com/github/codeql-go/commit/824b5a4b52b92859b4963f702b3c117e729b815e
because Browser would not send credentials in this case, see also:
https://github.com/github/codeql-go/pull/542#issuecomment-857600640

**Comment by smowton** [comment]
@AdmiralGaust does that make sense to you? OK to close this?

**Comment by AdmiralGaust** [comment]
Yeah, it makes sense. But the other variant also goes undetected which allows the origin from header value. 

```
import(
"github.com/go-chi/cors"
)

var corsOpts = cors.Options{
	AllowedOrigins:   []string{"http://*"},
	AllowCredentials: true
}

```
